src/preamble.js

```diff
@@ -932,6 +932,16 @@ function stackTrace() {
 }
 {{{ maybeExport('stackTrace') }}}
 
+function isAbsoluteUrl(url) {
+  return /^(?:[a-z]+:)?\/\//i.test(url);
+}
+
+function joinUrl(a, b) {
```

Comment:
please add a comment that this both joins as the name suggests, but also allows `a` to be undefined, in which case it is ignored.